Cleanup db/* references in smrt-server-analysis

CC @mpkocher @skinner 

Split out from this comment https://github.com/PacificBiosciences/smrtflow/issues/55#issuecomment-219853372. It seems we have some hard-coded path logic related to using an sqlite db.

Run the code with the below or directly from IntelliJ.

```bash
sbt smrt-server-analysis/assembly
cd smrt-server-analysis
mkdir db
java -jar target/scala-2.11/smrt-server-analysis*.jar
```

Output will error with similar to the following.

```bash
Connected to the target VM, address: '127.0.0.1:57556', transport: 'socket'
Warning. Assuming relative path /Users/jfalkner/tokeep/git/PacificBiosciences/smrtflow/jobs-root
[WARNING] Unable to load CMD template from pb-engine.pb-cmd-template Error No configuration setting found for key 'pb-engine.pb-cmd-template'
org.flywaydb.core.api.FlywayException: Unable to obtain Jdbc connection from DataSource (jdbc:sqlite:db/analysis_services.db) for user '': path to 'db/analysis_services.db': '/Users/jfalkner/tokeep/git/PacificBiosciences/smrtflow/db' does not exist
	at org.flywaydb.core.internal.util.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:407)
	at org.flywaydb.core.internal.util.jdbc.DriverDataSource.getConnection(DriverDataSource.java:365)
	at org.flywaydb.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:50)
	at org.flywaydb.core.Flyway.execute(Flyway.java:1314)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:917)
	at com.pacbio.secondary.smrtlink.actors.JobsDao.initializeDb(JobsDao.scala:935)
	at com.pacbio.secondary.smrtserver.appcomponents.SecondaryApi$class.startup(SecondaryAnalysisApp.scala:49)
	at com.pacbio.secondary.smrtserver.appcomponents.SecondaryAnalysisServer$.startup(SecondaryAnalysisApp.scala:67)
	at com.pacbio.common.app.BaseApi$ServiceActor.preStart(BaseSmrtServerApp.scala:140)
	at akka.actor.Actor$class.aroundPreStart(Actor.scala:470)
	at com.pacbio.common.services.package$RoutedHttpService.aroundPreStart(package.scala:169)
	at akka.actor.ActorCell.create(ActorCell.scala:580)
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:456)
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:478)
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:263)
	at akka.dispatch.Mailbox.run(Mailbox.scala:219)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.sql.SQLException: path to 'db/analysis_services.db': '/Users/jfalkner/tokeep/git/PacificBiosciences/smrtflow/db' does not exist
	at org.sqlite.core.CoreConnection.open(CoreConnection.java:117)
	at org.sqlite.core.CoreConnection.<init>(CoreConnection.java:66)
	at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:21)
	at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:23)
	at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:45)
	at org.sqlite.JDBC.createConnection(JDBC.java:114)
	at org.sqlite.JDBC.connect(JDBC.java:88)
	at org.flywaydb.core.internal.util.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:404)
	... 20 more
```

--- a/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
+++ b/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
@@ -36,8 +36,11 @@ class Dal(val dbURI: String) {
   val flyway = new Flyway() {
     override def migrate(): Int = {
       // lazy make file database dir as needed
-      val file = scala.reflect.io.File(dbURI.stripPrefix("jdbc:sqlite:"))
-      if (file.parent != null && !file.parent.exists) file.parent.createDirectory()
+      val file = Paths.get(dbURI.stripPrefix("jdbc:sqlite:"))
+      if (file.getParent != null) {
+        val dir = file.getParent.toFile
+        if (!dir.exists()) dir.mkdirs()
+      }
 
       return super.migrate()
     }

--- a/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
+++ b/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
@@ -28,8 +28,9 @@ import scala.language.postfixOps
 import scala.slick.driver.SQLiteDriver.simple._
 import scala.slick.jdbc.meta.MTable
 import scala.util.{Failure, Success, Try}
-
 import org.flywaydb.core.Flyway
+
+import scala.reflect.io.Directory
 
 
 // TODO(smcclellan): Move this class into the c.p.s.s.database package? Or eliminate it?
@@ -932,6 +933,10 @@ with DataSetStore {
   var _runnableJobs = mutable.Map[UUID, RunnableJobWithId]()
 
   def initializeDb(): Unit = {
+    // lazy make ./db as needed
+    val dir = Directory("db")
+    if (!dir.exists) dir.createDirectory()
+
     dal.flyway.migrate()
   }
 }

--- a/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
+++ b/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
@@ -35,8 +35,8 @@ import org.flywaydb.core.Flyway
 class Dal(val dbURI: String) {
   val flyway = new Flyway() {
     override def migrate(): Int = {
-      // lazy make dev `db` database dir as needed. intended only for dev work
-      if (dbURI == "jdbc:sqlite:db/analysis_services.db") {
+      // lazy make directories as needed for sqlite
+      if (dbURI.startsWith("jdbc:sqlite:")) {
         val file = Paths.get(dbURI.stripPrefix("jdbc:sqlite:"))
         if (file.getParent != null) {
           val dir = file.getParent.toFile

--- a/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
+++ b/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
@@ -30,12 +30,18 @@ import scala.slick.jdbc.meta.MTable
 import scala.util.{Failure, Success, Try}
 import org.flywaydb.core.Flyway
 
-import scala.reflect.io.Directory
-
 
 // TODO(smcclellan): Move this class into the c.p.s.s.database package? Or eliminate it?
 class Dal(val dbURI: String) {
-  val flyway = new Flyway()
+  val flyway = new Flyway() {
+    override def migrate(): Int = {
+      // lazy make file database dir as needed
+      val file = scala.reflect.io.File(dbURI.stripPrefix("jdbc:sqlite:"))
+      if (file.parent != null && !file.parent.exists) file.parent.createDirectory()
+
+      return super.migrate()
+    }
+  }
   flyway.setDataSource(dbURI, "", "")
   flyway.setBaselineOnMigrate(true)
   flyway.setBaselineVersionAsString("1")
@@ -933,10 +939,6 @@ with DataSetStore {
   var _runnableJobs = mutable.Map[UUID, RunnableJobWithId]()
 
   def initializeDb(): Unit = {
-    // lazy make ./db as needed
-    val dir = Directory("db")
-    if (!dir.exists) dir.createDirectory()
-
     dal.flyway.migrate()
   }
 }

--- a/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
+++ b/smrt-server-link/src/main/scala/com/pacbio/secondary/smrtlink/actors/JobsDao.scala
@@ -35,11 +35,13 @@ import org.flywaydb.core.Flyway
 class Dal(val dbURI: String) {
   val flyway = new Flyway() {
     override def migrate(): Int = {
-      // lazy make file database dir as needed
-      val file = Paths.get(dbURI.stripPrefix("jdbc:sqlite:"))
-      if (file.getParent != null) {
-        val dir = file.getParent.toFile
-        if (!dir.exists()) dir.mkdirs()
+      // lazy make dev `db` database dir as needed. intended only for dev work
+      if (dbURI == "jdbc:sqlite:db/analysis_services.db") {
+        val file = Paths.get(dbURI.stripPrefix("jdbc:sqlite:"))
+        if (file.getParent != null) {
+          val dir = file.getParent.toFile
+          if (!dir.exists()) dir.mkdirs()
+        }
       }
 
       return super.migrate()
@@ -374,7 +376,8 @@ trait JobDataStore extends JobEngineDaoComponent with LazyLogging {
 
   /**
    * This is the new interface will replace the original createJob
-   * @param uuid UUID
+    *
+    * @param uuid UUID
    * @param name Name of job
    * @param description This is really a comment. FIXME
    * @param jobTypeId String of the job type identifier. This should be consistent with the
@@ -478,7 +481,8 @@ trait DataSetStore extends DataStoreComponent with LazyLogging {
 
   /**
    * Importing of DataStore File by Job Int Id
-   * @param ds
+    *
+    * @param ds
    * @param jobId
    * @return
    */
@@ -503,7 +507,8 @@ trait DataSetStore extends DataStoreComponent with LazyLogging {
 
   /**
    * Generic Importing of DataSet by type and Path to dataset file
-   * @param dataSetMetaType
+    *
+    * @param dataSetMetaType
    * @param spath
    * @param jobId
    * @param userId

